apps/passport-server/src/services/generic-issuance/pipelines/PretixPipeline.ts

```diff
@@ -896,20 +931,31 @@ export class PretixPipeline implements BasePipeline {
       const { emailClaim } =
         await this.credentialSubservice.verifyAndExpectZupassEmail(req.pcd);
 
-      if ((this.definition.options.semaphoreGroups ?? []).length > 0) {
-        const didUpdate = await this.consumerDB.save(
-          this.id,
-          emailClaim.emailAddress,
-          emailClaim.semaphoreId,
-          new Date()
+      const didUpdate = await this.consumerDB.save(
+        this.id,
+        emailClaim.emailAddress,
+        emailClaim.semaphoreId,
+        new Date()
+      );
+
+      if (this.autoIssuanceProvider) {
+        const newManualTickets =
+          await this.autoIssuanceProvider.maybeIssueForUser(
+            emailClaim.emailAddress,
+            await this.getAllManualTickets(),
+            await this.db.loadByEmail(this.id, emailClaim.emailAddress)
+          );
+
+        await Promise.allSettled(
+          newManualTickets.map((t) => this.manualTicketDB.save(this.id, t))
         );
+      }
 
-        // If the user's Semaphore commitment has changed, `didUpdate` will be
-        // true, and we need to update the Semaphore groups
-        if (didUpdate) {
-          span?.setAttribute("semaphore_groups_updated", true);
-          await this.triggerSemaphoreGroupUpdate();
-        }
+      // If the user's Semaphore commitment has changed, `didUpdate` will be
+      // true, and we need to update the Semaphore groups
+      if (didUpdate) {
```

Comment:
Given that the check for whether Semaphore groups are configured was removed above, I think one should be added around this block.

My understanding is that previously we only saved consumer data if Semaphore groups were configured, and now we're saving that data in all cases. However, we don't want to trigger a Semaphore group update, even if consumer data has been updated, if there are no Semaphore groups configured.

Reply:
done